source/brailleDisplayDrivers/braillePen12.py

```diff
@@ -0,0 +1,182 @@
+#brailleDisplayDrivers/braillePen12.py
+#A part of NonVisual Desktop Access (NVDA)
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+#Copyright (C) 2014 Harpo Sp. z o. o. <http://www.harpo.com.pl>
+
+import braille
+import queueHandler
+from logHandler import log
+from ctypes import *
+from ctypes.wintypes import *
+import time
+import config
+import inputCore
+import hwPortUtils
+import brailleInput
+import wx
+
+READ_INTERVAL = 50
+
+BP12_KEYCALLBACKHANDLE = WINFUNCTYPE(None,c_ubyte,c_ubyte,c_ubyte,c_ubyte,c_ubyte,c_ubyte,c_ubyte,c_ubyte)
+
+BP12_BRAILLE_KEYS = ("dot1", "dot2", "dot3", "dot4", "dot5", "dot6", "dot7", "dot8")
+BP12_CONTROL_KEYS=("leftScroll","joyAction","joyLeft","joyUp","joyDown","joyRight","space", "rightScroll")
+
+#Try to load BP12TI32.dll
+try:
+	BP12Lib=windll[r"brailleDisplayDrivers\BP12TI.dll"]
+except:
+	BP12Lib=None
+
+def _findPorts():
+	for portInfo in hwPortUtils.listComPorts(onlyAvailable=True):
+		try:
+			btName = portInfo["bluetoothName"]
+		except KeyError:
+			continue
+		if btName.startswith("EL12-"):
+			yield portInfo["port"]
+
+class BrailleDisplayDriver(braille.BrailleDisplayDriver):
+	name = "braillePen12"
+	# Translators: The name of a series of braille displays.
+	description = _("BP12 and EL12 series")
+	isThreadSafe = True
+	
+	@classmethod
+	def check(cls):
+		return bool(BP12Lib)
```

Comment:
There's no manual port selection here, so this should return False if no usable ports were found.